test/test_large_snopt.py

```diff
@@ -59,83 +67,68 @@ def sens(self, xdict, funcs):
+    def setup_optProb(self, sparse=True):
         # set N
         if sparse:
             self.N = 50000
         else:
             self.N = 500
+
         # Optimization Object
-        optProb = Optimization("large and sparse", self.objfunc)
+        self.optProb = Optimization("large and sparse", self.objfunc, sens=self.sens)
 
         # Design Variables
-        optProb.addVar("x", lower=-100, upper=150, value=0)
-        optProb.addVarGroup("y", self.N, lower=-10 - np.arange(self.N), upper=np.arange(self.N), value=0)
-        optProb.addVarGroup("z", 2 * self.N, upper=np.arange(2 * self.N), lower=-100 - np.arange(2 * self.N), value=0)
+        self.optProb.addVar("x", lower=-100, upper=150, value=0)
+        self.optProb.addVarGroup("y", self.N, lower=-10 - np.arange(self.N), upper=np.arange(self.N), value=0)
+        self.optProb.addVarGroup(
+            "z", 2 * self.N, upper=np.arange(2 * self.N), lower=-100 - np.arange(2 * self.N), value=0
+        )
 
         # Constraints
-        optProb.addCon("con1", upper=100, wrt=["x"])
-        optProb.addCon("con2", upper=100)
-        optProb.addCon("con3", lower=4, wrt=["x", "z"])
+        self.optProb.addCon("con1", upper=100, wrt=["x"])
+        self.optProb.addCon("con2", upper=100)
+        self.optProb.addCon("con3", lower=4, wrt=["x", "z"])
         xJac = np.ones((self.N, 1))
         if sparse:
             yJac = scipy.sparse.spdiags(np.ones(self.N), 0, self.N, self.N)
         else:
             yJac = np.eye(self.N)
-        optProb.addConGroup(
+        self.optProb.addConGroup(
             "lincon",
             self.N,
             lower=2 - 3 * np.arange(self.N),
             linear=True,
             wrt=["x", "y"],
             jac={"x": xJac, "y": yJac},
         )
-        optProb.addObj("obj")
-
-        # Optimizer
-        try:
-            opt = SNOPT(options=optOptions)
-        except Error:
-            raise unittest.SkipTest("Optimizer not available: SNOPT")
-
-        sol = opt(optProb, sens=self.sens)
-
-        # Check Solution
-        if tol is not None:
-            if opt.version != "7.7.7":
-                assert_allclose(sol.objectives["obj"].value, 10.0, atol=tol, rtol=tol)
-            else:
-                assert_allclose(sol.fStar, 10.0, atol=tol, rtol=tol)
-            assert_allclose(sol.variables["x"][0].value, 2.0, atol=tol, rtol=tol)
-        return sol
+        self.optProb.addObj("obj")
 
     def test_sparse(self):
-        test_name = "large_sparse_SNOPT"
-        optOptions = {
-            "Print file": "{}.out".format(test_name),
-            "Summary file": "{}_summary.out".format(test_name),
-        }
-        self.optimize(tol=1e-5, optOptions=optOptions)
+        self.optName = "SNOPT"
+        self.setup_optProb(sparse=True)
+        sol = self.optimize()
+        self.assert_solution_allclose(sol, 1e-5, partial_x=True)
+
+    def test_sparse_IPOPT(self):
```

Comment:
yeah I agree, can't we use @parameterized(["SNOPT", "IPOPT"]) for `test_sparse` @nwu63 ?

Reply:
Good point, will rename the file and parameterize those. Also Marco because you didn't wrap that in quotes you just notified some random person on GitHub :stuck_out_tongue: 

I added IPOPT because I don't think we have been testing its sparse Jacobian capabilities, but maybe the problem size is a bit unreasonable. Should we reduce the size?

Reply:
I think the size could be reduced. I just checked the Azure logs and this one test is about 50% of the 'Run Tests' time.

Reply:
For some reason, setting it to `5000` resulted in some segfaults within IPOPT, while 10000 is okay. Could you double check and see if you could reproduce that on your local machine?

Reply:
Yes, I was able to reproduce the segfault. That is very strange.